### PR TITLE
Move source prettifying information to input file

### PR DESCRIPTION
**Pull request by nicarl**
Adding a new field called `externalAttributionSources` to the input file. This is a replacement for the hard coded mapping for attribution sources (e.g. `SC`) to a prettified name such as `ScanCode`. Furthermore the data structure contains a sorting key, which determines the sorting in the `PackagePanel`.

### CODE REVIEW

File: src/Frontend/Components/PackagePanel/package-panel-helpers.ts [hunk 1]
```diff
@@ -43,22 +30,32 @@ export function getSortedPrettifiedSources(
...
+  const sortedExistingKnownSources = sources
+    .reduce((existingKnownSources: Array<string>, source: string) => {
+      if (attributionSources.hasOwnProperty(source)) {
+        existingKnownSources.push(source);
+      } else {
+        existingUnknownSources.push(source);
+      }
+      return existingKnownSources;
+    }, [])
+    .sort((sourceA, sourceB) => {
```
Comment (nicarl) [inline src/Frontend/Components/PackagePanel/package-panel-helpers.ts:51]: Sorting should be also alphabetically if priorities are identical.

File: src/ElectronBackend/input/externalAttributionSources.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// SPDX-FileCopyrightText: Facebook, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { ExternalAttributionSources } from '../../shared/shared-types';
+
+export const ATTRIBUTION_SOURCES: ExternalAttributionSources = {
```
Comment (sebathomas) [inline src/ElectronBackend/input/externalAttributionSources.ts:8]: Can you add a TODO to drop support at some point?

File: example-files/opossum_input.json [hunk 3]
```diff
@@ -228,5 +228,11 @@
   "baseUrlsForSources": {
     "/": "https://github.com/opossum-tool/opossumUI/{path}?test=87583cf6c5d859bef90ee2b37fffe7a485c4d7a2",
     "/Frontend/": "https://github.com/opossum-tool/opossumUI/{path}?test=87583cf6c5d859bef90ee2b37fffe7a485c4d7a4"
+  },
+  "externalAttributionSources": {
+    "SC": {
+      "name": "Scancode",
```
Comment (sebathomas) [inline example-files/opossum_input.json:234]: ScanCode

File: src/Frontend/Components/PackagePanel/package-panel-helpers.ts (hunk 1 above)
Comment (sebathomas) [reply src/Frontend/Components/PackagePanel/package-panel-helpers.ts:51]: That's still missing?

File: src/Frontend/Components/PackagePanel/package-panel-helpers.ts [hunk 4]
```diff
@@ -43,22 +30,32 @@ export function getSortedPrettifiedSources(
...
-    .filter(
-      (attributionSource) =>
-        !ATTRIBUTION_SOURCES_ORDER.includes(attributionSource)
-    )
-    .sort();
+function sortSources(
+  sources: Array<string>,
+  attributionSources: ExternalAttributionSources
+): Array<string> {
+  const existingUnknownSources: Array<string> = [];
```
Comment (sebathomas) [inline src/Frontend/Components/PackagePanel/package-panel-helpers.ts:40]: Seems weird to have one list in the reduce and another list outside. I think it should be symmetrical.

**Review by sebathomas — APPROVED** [review]

**Review by antonbauhofer — APPROVED** [review]
Looks good to me.

File: src/ElectronBackend/input/OpossumInputFileSchema.json [hunk 5]
```diff
@@ -204,6 +204,26 @@
       "type": "object",
       "description": "Map from paths to the respective base url. The base url should contain a {path} placeholder. E.g. https://github.com/opossum-tool/opossumUI/blob/main/{path}",
       "additionalProperties": true
+    },
+    "externalAttributionSources": {
+      "type": "object",
+      "description": "",
```
Comment (antonbauhofer) [inline src/ElectronBackend/input/OpossumInputFileSchema.json:210]: We could add a description here.

File: src/ElectronBackend/input/externalAttributionSources.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  'REUSER:HC': { name: 'High Compute (old scan)', priority: 7 },
+  SC: { name: 'ScanCode', priority: 8 },
+  HC: { name: 'High Compute', priority: 9 },
+  HINT: { name: 'Hint', priority: 10 },
+};
+
+export function combineExternalAttributionSources(
+  sources: Array<ExternalAttributionSources>
+): ExternalAttributionSources {
+  if (sources.length == 0) {
```
Comment (benedikt-richter) [inline src/ElectronBackend/input/externalAttributionSources.ts:25]: === should be preferred vs ==

File: src/Frontend/Components/PackagePanel/__tests__/package-panel-helpers.test.ts [hunk 7]
```diff
@@ -101,26 +103,64 @@ describe('PackagePanel helpers', () => {
...
+      HINT: { name: 'Hint', priority: 1 },
+    };
+    expect(
+      getSortedSources(
+        testAttributions,
+        testAttributionIds,
+        testAttributionSources
+      )
+    ).toEqual([
+      'HC', // High Compute
```
Comment (benedikt-richter) [inline src/Frontend/Components/PackagePanel/__tests__/package-panel-helpers.test.ts:157]: why do we have these comments here?

File: src/Frontend/integration-tests/__tests__/other-popups.test.tsx [hunk 8]
```diff
@@ -40,6 +40,8 @@ import { App } from '../../Components/App/App';
 
 import { TIME_POPUP_IS_DISPLAYED } from '../../Components/ErrorPopup/ErrorPopup';
 import { ButtonTitle, DiscreteConfidence } from '../../enums/enums';
+import { setExternalAttributionSources } from '../../state/actions/resource-actions/all-views-simple-actions';
+import { ATTRIBUTION_SOURCES } from '../../../ElectronBackend/input/externalAttributionSources';
```
Comment (benedikt-richter) [inline src/Frontend/integration-tests/__tests__/other-popups.test.tsx:44]: we should not import from the BE into the FE

File: src/Frontend/state/actions/resource-actions/__tests__/load-actions.test.ts [hunk 9]
```diff
@@ -31,6 +32,7 @@ import {
 import { getResolvedExternalAttributions } from '../../../selectors/audit-view-resource-selectors';
 import { loadFromFile } from '../load-actions';
 import { EMPTY_PROJECT_METADATA } from '../../../../shared-constants';
+import { ATTRIBUTION_SOURCES } from '../../../../../ElectronBackend/input/externalAttributionSources';
```
Comment (benedikt-richter) [inline src/Frontend/state/actions/resource-actions/__tests__/load-actions.test.ts:35]: as above

**Review by benedikt-richter — APPROVED** [review]
just minor points

**Pull request merged**